mobile/test/java/io/envoyproxy/envoymobile/engine/testing/XdsTestServerFactory.java

```diff
@@ -0,0 +1,42 @@
+package io.envoyproxy.envoymobile.engine.testing;
+
+/** An xDS test server factory. */
+public class XdsTestServerFactory {
+  /** The instance of {@link XdsTestServer}. */
+  public static class XdsTestServer {
```

Comment:
I'm wondering if it would be better to have the XdsTestServer as a top-level class, outside of XdsTestServerFactory?  Same for the other HTTP servers from previous PRs.  I think it'll make the code easier to read, but just a personal preference.  I consider the XdsTestServer to be a top level construct, not dependent on XdsTestServerFactory. It's just that XdsTestServerFactory gives us XdsTestServer instances

Reply:
Unfortunately it's not possible to do it in Java unless we create two separate files. Having a single file is better IMO.

Reply:
ah ok, yeah agree, this is better then, thanks for the explanation!